Teacher Workout/Features/Home/Coordinator/HomeCoordinator.swift

```diff
@@ -23,7 +23,36 @@ extension HomeCoordinator: HomeViewDelegate {
     func homeView(_: HomeView, didSelectTheme theme: Theme) {
         let viewModel = ThemeDetailsViewModel(theme: theme)
         let viewDetails = ThemeDetails(viewModel: viewModel)
+
+        viewModel.lessonSelected = { [weak self] lesson in
+            guard let self = self else { return }
+
+            let introView = LessonIntroView(lesson: lesson, delegate: self)
+            let viewController = UIHostingController(rootView: introView)
+            self.navigationController.present(viewController, animated: true)
+        }
+
         let viewController = UIHostingController(rootView: viewDetails)
         navigationController.pushViewController(viewController, animated: true)
     }
+
+    func homeView(_: HomeView, didSelectLesson lesson: Lesson) {
+        let introView = LessonIntroView(lesson: lesson, delegate: self)
+        let viewController = UIHostingController(rootView: introView)
+        navigationController.present(viewController, animated: true)
+    }
+}
+
+extension HomeCoordinator: LessonIntroViewDelegate {
+    func lessonIntroViewDidTapClose(_: LessonIntroView) {
+        navigationController.dismiss(animated: true)
+    }
+
+    func lessonIntroViewDidTapStartLesson(_: LessonIntroView) {
+        navigationController.dismiss(animated: true)
+    }
+
+    func lessonIntroViewDidTapSaveLesson(_: LessonIntroView) {
+        navigationController.dismiss(animated: true)
+    }
```

Comment:
La fel si partea asta este asemanatoare cu `DiscoverCoordinator`

Reply:
`DiscoverCoordinator` si `HomeCoordinator` implementeaza acelasi delegat deci functiile implementate sunt aceleasi. Momentan pare ca fac acelasi lucru, dar poate in viitor DiscoverCoordinator va face ceva cu ele, HomeCoordinator altceva, le-as lasa asa, pare mai readable.